File: tests/fixtures/vulnerable/authorization_test.py
# aiscan test fixture — should trigger AI-SEC-002 (Missing Authorization)
# DO NOT USE IN PRODUCTION

from flask import Flask, jsonify
from fastapi import FastAPI

app = Flask(__name__)
api = FastAPI()


# VULNERABLE 1: Flask admin route with no auth decorator
@app.route('/admin/users')
def admin_users():
    return jsonify([])


# VULNERABLE 2: Flask DELETE mutation with no auth
@app.route('/api/users/<int:user_id>', methods=['DELETE'])
def delete_user(user_id):
    return jsonify({'deleted': user_id})


# VULNERABLE 3: Flask POST mutation on sensitive path
@app.post('/api/posts')
def create_post():
    return jsonify({'id': 1}), 201


# VULNERABLE 4: FastAPI user-specific data with no Depends
@api.get('/user/{user_id}/profile')
async def get_user_profile(user_id: int):
    return {'user_id': user_id}


# VULNERABLE 5: FastAPI admin endpoint
@api.delete('/admin/posts/{post_id}')
async def admin_delete_post(post_id: int):
    return {'deleted': post_id}


# SAFE 1: @login_required present — should NOT be flagged
@app.route('/admin/dashboard')
@login_required  # noqa: F821
def admin_dashboard():
    return jsonify({'status': 'ok'})


# SAFE 2: FastAPI Depends — should NOT be flagged
@api.get('/user/{user_id}/data')
async def get_user_data(user_id: int, current_user=Depends(get_current_user)):  # noqa: F821
    return {'user_id': user_id}


# SAFE 3: Public health endpoint — should NOT be flagged
@app.get('/health')
def health():
    return {'status': 'ok'}


# SAFE 4: Public read endpoint — should NOT be flagged
@app.get('/api/public/articles')
def list_articles():
    return jsonify([])

File: tests/fixtures/vulnerable/authorization_test.js
// aiscan test fixture — should trigger AI-SEC-002 (Missing Authorization)
// DO NOT USE IN PRODUCTION

const express = require('express');
const app = express();
const router = express.Router();

// VULNERABLE 1: Express admin route with no middleware
app.get('/admin/dashboard', (req, res) => {
    res.json({ users: [] });
});

// VULNERABLE 2: Express DELETE with no auth
router.delete('/api/user/:id', async (req, res) => {
    await User.findByIdAndDelete(req.params.id);
    res.json({ deleted: req.params.id });
});

// VULNERABLE 3: Express POST on sensitive path
app.post('/api/posts', async (req, res) => {
    const post = await Post.create(req.body);
    res.status(201).json(post);
});

// VULNERABLE 4: User-specific data access, no auth check
router.get('/user/:userId/settings', (req, res) => {
    res.json({ userId: req.params.userId });
});

// VULNERABLE 5: Payment endpoint no auth
app.post('/api/payment/charge', async (req, res) => {
    res.json({ charged: true });
});

// SAFE 1: Named middleware in argument position — should NOT be flagged
app.get('/admin/reports', requireAuth, (req, res) => {
    res.json({ reports: [] });
});

// SAFE 2: Manual auth check in handler — should NOT be flagged
app.delete('/api/comment/:id', async (req, res) => {
    if (!req.user) return res.status(401).json({ error: 'Unauthorized' });
    res.json({ deleted: req.params.id });
});

// SAFE 3: Public endpoint — should NOT be flagged
app.get('/api/public/posts', (req, res) => {
    res.json([]);
});

// SAFE 4: Health check — should NOT be flagged
app.get('/health', (req, res) => {
    res.json({ status: 'ok' });
});
